documentation/docs/getting-started/core-concepts.md

```diff
@@ -0,0 +1,153 @@
+# Core Concepts
+
+### Middlewares
```

Comment:
Feels like `Middleware` by itself is already plural, maybe we should drop the 's' here and in other places in this PR?

Reply:
I guess that makes sense, we don't say "softwares" after all!